src/complete.cpp

```diff
@@ -224,13 +224,13 @@ completion_t &completion_t::operator=(const completion_t &him) {
 }
 
 bool completion_t::is_naturally_less_than(const completion_t &a, const completion_t &b) {
+    //for this to work, stable_sort must be used because results aren't interchangeable
+    if ((a.flags & COMPLETE_DONT_SORT) != 0 && (b.flags & COMPLETE_DONT_SORT) != 0) {
```

Comment:
Why is it necessary to check both a and b? Since we're sorting the values for a completion shouldn't they have the same flags? Also, no need for the explicit `!= 0` test.